hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -200,6 +202,41 @@ public void run() {
         }, 1, 1, TimeUnit.SECONDS);
     }
 
+    private void openConnectionsToMembers() {
+        ClientClusterService clientClusterService = client.getClientClusterService();
+        Collection<Member> memberList = clientClusterService.getMemberList();
+        long startTimeMillis = System.currentTimeMillis();
+        while (true) {
+            Exception exception = null;
+            for (Member member : memberList) {
+                try {
+                    clientConnectionManager.getOrConnect(member.getAddress());
+                } catch (Exception e) {
+                    if (!clientClusterService.getMemberList().contains(member)) {
+                        //member left the cluster no need to connect
+                        continue;
+                    }
+                    exception = e;
```

Comment:
will it continue? then we end up overwriting the previous exception.

Reply:
I did not see any problem with overwriting previous exception. Any non - null exception reported in cause I think is enough. 
Otherwise I need to keep list of exceptions.User needs to have a getCauses method on OperationTimeoutException, and they don't .

Reply:
Or you can break the loop on the first exception, it is up to you.